### DISCUSSION TITLE
Fun with ippeveprinter

### DISCUSSION THREAD

**Discussion by debiantriage** [OpenPrinting/cups#770] [Ideas]
cups-pdf is available in Debian and other distributions. A facet of its operation is that every submitted file (even a PDF) is first converted to PostScript, which is then converted to PDF. This might be seen as suboptimal.

ippeveprinter  runs  a  command or a script when a job to be printed enters its spool directory, so let's execute
```
/usr/sbin/ippeveprinter-c $HOME/pdfsave -f application/pdf "Print to PDF" &
```
The script pdfsave is
```
#!/bin/bash
mkdir -p $HOME/PDF/
A=$(pdfinfo "$1" | grep texttopdf)
if [ ! -z "$A" ]
then
   /usr/bin/pdftocairo -pdf "$1" $HOME/PDF/$(basename $1)
else
   cp $1 $HOME/PDF
fi
```
PDFs that have been processed by the texttopdf filter are not searchable; pdftocairo makes them so.

The Print_to_PDF printer is, of course, accessible on the local network.  Should this not be wanted, a firewall rule might be the answer:
```
iptables -I INPUT \! --src 192.168.7.44 -m tcp -p tcp --dport 8001 -j DROP
```
Unlike an application's Print to File feature, Print_to_PDF can be used for batch printing with `lp`.

**Comment by gmuth** [comment]
Great idea. However I have difficulties getting ippeveprinter to run in docker:
```
docker run --rm anujdatar/cups ippeveprinter "Everywhere"
Error: Unable to initialize DNS-SD.
```
